ecosystem/python/sdk/aptos_sdk/client.py

```diff
@@ -162,11 +162,15 @@ def transaction_pending(self, txn_hash: str) -> bool:
     def wait_for_transaction(self, txn_hash: str) -> None:
         """Waits up to 10 seconds for a transaction to move past pending state."""
 
-        count = 0
+        count_start = time.time()
+        prev_time = round(time.time() / 0.2)
         while self.transaction_pending(txn_hash):
-            assert count < 10, f"transaction {txn_hash} timed out"
-            time.sleep(1)
-            count += 1
+            t = round(time.time() / 0.2)
+            if prev_time == t: # To reduce the rest server load and polite access rest server.
+                time.sleep(0.2)
+            else:
+                prev_time = t
+            assert count_start + 10 > time.time(), f"transaction {txn_hash} timed out"
```

Comment:
This is a very interesting proposal.

But `prev_time` and `t` will almost never be equal.

I would suggest just making a variable, say Client::txn_backoff and set it by default to 1. Then after you create client, you can just set it to whatever you want.